oslo_vmware/rw_handles.py

```diff
@@ -308,10 +313,38 @@ def __init__(self, session, lease, url, file_handle,
 
     def _log_progress(self, progress):
         """Log data transfer progress."""
-        if (progress == 100 or (progress - self._last_logged_progress >=
-                                MIN_PROGRESS_DIFF_TO_LOG)):
-            LOG.debug("Data transfer progress is %d%%.", progress)
-            self._last_logged_progress = progress
+        time_diff = time.time() - self._last_progress_update
+        progress["time_diff"] = datetime.timedelta(seconds=time_diff)
```

Comment:
I see why you do this, but with doing so `_log_progress()` changes its input `progress` which is still used by another function later on. This can lead to unexpected behavior in the long run.

Reply:
Fixed